tools/create_github_release.py

```diff
@@ -0,0 +1,90 @@
+
+Setup:
+ - Create an azure personal access token
+   - https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate
+   - You'll need read scope
+   - Save the token to somewhere like ~/.ssh/azurepat.txt
+ - Create a github personal access token
+   - https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token#creating-a-token
+   - You'll need repo scope
+   - Save the token to somewhere like ~/.ssh/githubpat.txt
+
+Run:
+
+python tools/create_github_release.py 1.7.0 ~/.ssh/azurepat.txt ~/.ssh/githubpat.txt
+"""
+
+import requests
+import sys
+import tempfile
+import urllib.parse
+from zipfile import ZipFile
+
+from azure.devops.connection import Connection
+from github import Github
+from msrest.authentication import BasicAuthentication
+
+def download_azure_artifacts(azure_access_token, tempdir):
+    """Download and unzip build artifacts from Azure pipelines.
+
+    :param str azure_access_token: string containing azure access token
+    :param str path: path to a temporary directory to save the files
+
+    """
+    # Create a connection to the azure org
+    organization_url = 'https://dev.azure.com/certbot'
+    credentials = BasicAuthentication('', azure_access_token)
+    connection = Connection(base_url=organization_url, creds=credentials)
+
+    # Find the build artifacts
+    build_client = connection.clients.get_build_client()
+    get_builds_response = build_client.get_builds('certbot', definitions='3')
+    build_id = get_builds_response.value[0].id
+    artifacts = build_client.get_artifacts('certbot', build_id)
+
+    # Save and unzip files
+    for filename in ('windows-installer', 'changelog'):
+        url = build_client.get_artifact('certbot', build_id, filename).resource.download_url
+        r = requests.get(url)
+        with open(tempdir + '/' + filename + '.zip', 'wb') as f:
+            f.write(r.content)
+        with ZipFile(tempdir + '/' + filename + '.zip', 'r') as zipObj:
+           zipObj.extractall(tempdir)
+
+def create_github_release(github_access_token, tempdir, version):
+    """Download and unzip build artifacts from Azure pipelines.
+
+    :param str github_access_token: string containing github access token
+    :param str path: path to a temporary directory where azure artifacts are located
+    :param str version: Certbot version number, e.g. 1.7.0
+
+    """
+    # Create release
+    g = Github(github_access_token)
+    repo = g.get_user('certbot').get_repo('certbot')
+    release_notes = open(tempdir + '/changelog/release_notes.md', 'r').read()
+    release= repo.create_git_release('v{0}'.format(version),
+                                     'Certbot {0}'.format(version),
+                                     release_notes)
+
+    # Upload windows installer to release
+    release.upload_asset(tempdir + '/windows-installer/certbot-beta-installer-win32.exe')
+
+def main(args):
+    version = args[0]
```

Comment:
In the spirit of making incremental improvements, I don't want to hold this PR up on cool things we could do too much but I had idea of something I think we could pretty easily do and I think it's a fairly significant improvement.

The time we had problems creating the GitHub release manually we typed the tag wrong which I think can still happen here. I looked into it a bit and I don't think getting the tag from Azure is that hard.
```
ipdb> build_client.get_build('certbot', build_id).source_branch                                                                                                                                                                                                                                                                                                                                                                       
'refs/tags/v1.7.0'
```
What do you think about making this call in this script and using the tag value there? We'd have to strip off the leading `refs/tags/` and remove the `v` when we don't want it rather than adding it, but I think we can pretty reliably do this. We do something similar [here](https://github.com/certbot/certbot/blob/220cc07239e9c5a7058640760ff79e738645b08e/.azure-pipelines/templates/steps/tox-steps.yml#L41).

What do you think? Do you think we should make this change now?

Reply:
oh, I love that!!! yeah, that should be super easy and extremely worth it.